frame/ethereum/src/lib.rs

```diff
@@ -210,6 +210,8 @@ decl_module! {
 
 			Pending::append((transaction, status, receipt));
 
+			Self::handle_exec(exit_reason)?;
```

Comment:
We actually still cannot return error here. It must succeed, because at this stage the runtime state has already changed.